eng/scripts/Update-VcpkgPort.ps1

```diff
@@ -0,0 +1,150 @@
+    exit 1
+} 
+
+if (!(Test-Path $ReleaseArtifactSourceDirectory/package-info.json)) { 
+    LogError "Could not locate package-info.json in -ReleaseArtifactSourceDirectory"
+    exit 1
+}
+
+if (!(Test-Path $ReleaseArtifactSourceDirectory/CHANGELOG.md)) { 
+    LogError "Could not locate CHANGELOG.md in -ReleaseArtifactSourceDirectory"
+}
+
+if (!(Test-Path $ReleaseArtifactSourceDirectory/vcpkg/port)) { 
+    LogError "Could not locate vcpkg/port directory in -ReleaseArtifactSourceDirectory"
+}
+
+# Clean out the folder so that template files removed are not inadvertently 
+# re-added
+if (Test-Path $PortDestinationDirectory) {
+    Remove-Item -v -r $PortDestinationDirectory
+}
+
+New-Item -Type Directory $PortDestinationDirectory
+Copy-Item `
+    -Verbose `
+    "$ReleaseArtifactSourceDirectory/vcpkg/port/*" `
+    $PortDestinationDirectory
+
+# Show artifacts copied into ports folder for PR
+Write-Host "Files in destination directory:" 
+Get-ChildItem -Recurse $PortDestinationDirectory | Out-String | Write-Host
+
+Write-Host "git status"
+git status
+
+# Temporarily commit changes to generate git tree objects required by 
+# "vcpkg x-add-version <port-name>"
+Write-Host "git add -A"
+git add -A
+Write-Host "git $GitCommitParameters commit -m 'Temporary commit to reset after x-add-version'"
+"git $GitCommitParameters commit -m 'Temporary commit to reset after x-add-version'" | Invoke-Expression -Verbose | Write-Host
+
+
+Write-Host "./bootstrap-vcpkg.bat"
+./bootstrap-vcpkg.bat
+
+if ($LASTEXITCODE -ne 0) { 
+    Write-Error "Failed to run bootstrap-vcpkg.bat"
+    exit 1
+}
+
+Write-Host "./vcpkg.exe x-add-version $VcpkgPortName"
+./vcpkg.exe x-add-version $VcpkgPortName
+
+if ($LASTEXITCODE -ne 0) { 
+    Write-Error "Failed to run vcpkg x-add-version $VcpkgPortName"
+    exit 1
+}
+
+# Reset to undo previous commit and put changes in the working directory.
+Write-Host "git reset HEAD^"
+git reset HEAD^
+
+# Grab content needed for commit message and place in a temporary file
+$packageVersion = (Get-Content $ReleaseArtifactSourceDirectory/package-info.json -Raw | ConvertFrom-Json).version
+$commitMessageFile = New-TemporaryFile
+$chagelogEntry = Get-ChangeLogEntryAsString `
+    -ChangeLogLocation $ReleaseArtifactSourceDirectory/CHANGELOG.md `
+    -VersionString $PackageVersion
+
+"[$VcpkgPortName] Update to $PackageVersion`n$chagelogEntry" `
+    | Set-Content $commitMessageFile
+
+Write-Host "Commit Message:"
+Write-host (Get-Content $commitMessageFile -Raw)
+
+
+Write-Host "git add -A"
+git add -A
+
+# Final commit using commit message from the temporary file. Using the file
+# enables the commit message to be formatted properly without having to write
+# code to escape certain characters that might appear in the changelog file.
+Write-Host "git $GitCommitParameters commit --file $commitMessageFile"
+"git $GitCommitParameters commit --file $commitMessageFile" `
```

Comment:
You did the revert but then you are commiting again? I assume this is because you cannot easily pass the commit message to the existing scripts? What if we enabled those steps to accept a file as the commit message on top of the existing string? Would that prevent any more need to do commits directly from here?

Reply:
Correct. We cannot easily read a multiline expression out into a variable without doing some escaping. Rather than write escaping code which could fail in the middle of a publish step in the future because it omits a case, I'd rather be safe and use the `--file` approach and skip the additional complexity. 

Git will error if you attempt to supply this as an extra paramter and use both `--file` and `-m` in the same command line.